business/mesh.go

```diff
@@ -262,20 +275,16 @@ func findKialiInNamespace(namespace string, clusterName string, clientSet kubern
 	if kialiNs != nil {
 		// The operator and the helm charts set this fixed label. It's also
 		// present in the Istio addon manifest of Kiali.
-		services, getSvcErr := clientSet.GetServicesByLabels(kialiNs.Name, "app.kubernetes.io/name=kiali")
+		services, getSvcErr := clientSet.GetServicesByLabels(kialiNs.Name, "app.kubernetes.io/part-of=kiali")
```

Comment:
@israel-hdez I'd like to ask you a change, perhaps not directly related to this PR but I guess it's important to introduce it or we'll add more technical debt to address.

In Kiali, all communication to the Kubernetes API should be done through the cache for local clusters, with the exceptions to some queries that doesn't have a proper Informer to build a cache.

Otherwise, we will introduce easily performance penalties very easily.

I understand that at the moment, we won't cache queries that are performed to a remote cluster, that's fine for now, but I'd ask if we can identify them in a better way, like renaming "clienteSet" to something like "remoteCluster" or similar.

That will help us to check if something is missing.

Reply:
Well, in this particular line it does not make sense a renaming. Here, the `clientSet` can be either a local or a remote client, depending on the caller. I think the clientSet is already abstracted from that fact.

However, I did a renaming on line 308 and also on line 487, it makes more sense on those lines.

Reply:
I'm not agree.
When I read this code, I interpret "you are calling a direct k8s without using any informer/cache, why ?"

In a local cluster we must not do that but using the cache, in a remote cluster it's ok, as we don't want to apply informers in that api yet.

So, if this function is intended to be used remotely,

```
func findKialiInNamespace(namespace string, clusterName string, clientSet kubernetes.ClientInterface) 
```

my suggestion is to change the "clientSet" to "remoteClientSet" or something that clarify that, otherwise, these kind of calls are a performance penalty that we need to review and be sure that caching techniques are not bypassed for other features.

Reply:
I haven't reviewed in deep this mesh.go class, but this is what it should be changed:

```
	// Discover ourselves
	kialiInstances := findKialiInNamespace(os.Getenv("ACTIVE_NAMESPACE"), myClusterName, in.k8s)
	if len(kialiInstances) > 0 && r != nil {
		for i := range kialiInstances {
			// If URL is already populated (because of an annotation), trust that because it's user configuration.
			if len(kialiInstances[i].Url) == 0 {
				kialiInstances[i].Url = httputil.GuessKialiURL(r)
			}
		}
	}
```
If that call is local, it's bypassing the services that are already cached.

You could fetch them from the cache and apply a local filter for labels, that would be better than perform a k8s call out of the cache scope.

Reply:
+1, I thought this was covered in Lucas' original comments, but local cluster queries should utilize the k8s cache.

Reply:
I wish the cache could implement our `kubernetes.ClientInterface` and encapsulate all this if/else logic :-/

A added a commit that should solve this.

Reply:
> I wish the cache could implement our `kubernetes.ClientInterface` and encapsulate all this if/else logic :-/

Not knowing most of this codebase yet... that sounds like a good refactor. The client implementation (or a separate `CacheReader` impl) could do cache reads by default and the callers wouldn't need to worry about when to use the cache. If you still wanted the option of reading from the API server you could add an `APIReader()` that just returned a non cache reading impl. That way callers would need to opt into non-cache reads and it would be easier to audit. Obviously out of scope for this PR though.